lib/open_project/plugins/module_handler.rb

```diff
@@ -0,0 +1,47 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2014 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++project_module
+
+module OpenProject::Plugins
+  module ModuleHandler
+    def self.disable_module(module_name)
+      project_modules =  Redmine::AccessControl.available_project_modules
+
+      binding.pry
+      if project_modules.include? (module_name.to_sym)
+        permissions = Redmine::AccessControl.permissions - Redmine::AccessControl.modules_permissions(module_name)
+        Redmine::AccessControl.map do |map|
+          permissions.map do |perm|
+            options = { project_module: perm.project_module, public: perm.public?, require: perm.require_loggedin? }
+            map.permission(perm.name, perm.actions, options)
+          end
+        end
+      end
+    end  
```

Comment:
I think you need to delay the actual removal of the permissions to a very late step in the initialisation. AFAIK, the initialisers of a mounted gem (plugin) are executed before the application's initialisers. Therefore, I'd try to use the ```to_prepare``` step. Please see the [documentation esp. Chapter 6](http://guides.rubyonrails.org/configuring.html) for details.

<pre>

class_attribute :disabled_modules

self.disabled_modules = []

def self.disable_module(module_name)
  disabled_modules << module_name
end

OpenProject::Application.config.to_prepare do
  [call method actually removing the permission with stored modules to remove]
end
 
</pre>

There is one drawback to this. In the development ENV, the initialisers are run but once whereas config.to_prepare is run before every request. The removal would only have to be run once, though. That will slow down the application in dev mode but it should't be to bad. It has no negative impact in production.